crates/symbolicator/src/services/symbolication.rs

```diff
@@ -1768,6 +1859,113 @@ fn load_cfi_for_processor(cfi: BTreeMap<DebugId, PathBuf>) -> BTreeMap<DebugId,
         .collect()
 }
 
+struct TempSymbolProvider {
+    files: BTreeMap<DebugId, SymbolFile>,
+    missing_ids: RefCell<BTreeSet<DebugId>>,
+}
+
+impl TempSymbolProvider {
+    /// Load the CFI information from the cache.
+    ///
+    /// This reads the CFI caches from disk and returns them in a format suitable for the
+    /// breakpad processor to stackwalk.
+    pub fn new<'a, M: Iterator<Item = (&'a DebugId, &'a PathBuf)>>(modules: M) -> Self {
+        // TODO(ja): Make TempSymbolProvider the thing serialized to procspawn (prepares for moving in-process)
+        Self {
+            files: modules
+                .filter_map(|(id, path)| Some((*id, Self::load(path)?)))
+                .collect(),
+
+            missing_ids: RefCell::new(BTreeSet::new()),
+        }
+    }
+
+    fn load(cfi_path: &Path) -> Option<SymbolFile> {
+        let bytes = ByteView::open(cfi_path)
+            .map_err(|err| {
+                let stderr: &dyn std::error::Error = &err;
+                tracing::error!(stderr, "Error while reading cficache");
+            })
+            .ok()?;
+
+        let cfi_cache = CfiCache::from_bytes(bytes)
+            // This mostly never happens since we already checked the files
+            // after downloading and they would have been tagged with
+            // CacheStatus::Malformed.
+            .map_err(|err| {
+                let stderr: &dyn std::error::Error = &err;
+                tracing::error!(stderr, "Error while loading cficache");
+            })
+            .ok()?;
+
+        SymbolFile::from_bytes(cfi_cache.as_slice())
+            .map_err(|err| {
+                let stderr: &dyn std::error::Error = &err;
+                tracing::error!(stderr, "Error while procecssing cficache");
+            })
+            .ok()
+    }
+
+    fn missing_ids(&self) -> BTreeSet<DebugId> {
+        std::mem::take(&mut *self.missing_ids.borrow_mut())
```

Comment:
We might as well call this `into_missing_ids(self)`, that would also solve the issue highlighted below.